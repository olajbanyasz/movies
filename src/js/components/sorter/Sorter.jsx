import React, { Component } from 'react';
import { ButtonToolbar, ToggleButtonGroup, ToggleButton } from 'react-bootstrap';
import './sorter.style.less';

class Sorter extends Component {

  state = {
    sortby: this.props.sortby
  };

  changeHandler = () => {
    if (event.target.value && event.target.value !== this.state.sortby) {
      this.setState({sortby: event.target.value}, () => this.props.changeHandler(this.state.sortby));
    }
  };

  render () {
    return (
      <div className="sorter">
        <span>Sort by</span>
        <ButtonToolbar className='sorter-buttons'>
          <ToggleButtonGroup
           type="radio"
           name='sorter'
           onChange={this.changeHandler}
           defaultValue={this.state.sortby}
          >
            <ToggleButton value={'DATE'} className='sort-by-date'>release date</ToggleButton>
            <ToggleButton value={'RATING'} className='sort-by-rating'>ratings</ToggleButton>
          </ToggleButtonGroup>
        </ButtonToolbar>
      </div> 
    );
  }
}

export default Sorter;