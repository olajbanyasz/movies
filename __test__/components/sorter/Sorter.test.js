import React from 'react';
import Sorter from '../../../src/js/components/sorter/Sorter.jsx';
import { render } from 'enzyme';

describe('Sorter', () => {
  it('should rendered correctly', () => {
    const component = render(<Sorter sortby={'DATE'} changeHandler={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});