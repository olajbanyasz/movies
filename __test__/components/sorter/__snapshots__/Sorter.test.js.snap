// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorter should rendered correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sorter",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Sort by",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sorter-buttons btn-toolbar",
            "role": "toolbar",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "sorter-button-group btn-group",
                "data-toggle": "buttons",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "sort-by-date btn btn-default active",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "sorter",
                        "type": "radio",
                        "value": "DATE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "release date",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "release date",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "sorter",
                          "type": "radio",
                          "value": "DATE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "sort-by-rating btn btn-default",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "sorter",
                          "type": "radio",
                          "value": "RATING",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "data": "ratings",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "text",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      Object {
                        "data": "ratings",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "autocomplete": "off",
                            "name": "sorter",
                            "type": "radio",
                            "value": "RATING",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "autocomplete": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "autocomplete": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                        },
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "sort-by-rating btn btn-default",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "sorter",
                        "type": "radio",
                        "value": "RATING",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "ratings",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "ratings",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "sorter",
                          "type": "radio",
                          "value": "RATING",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "sort-by-date btn btn-default active",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "sorter",
                          "type": "radio",
                          "value": "DATE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "data": "release date",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "text",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      Object {
                        "data": "release date",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "autocomplete": "off",
                            "checked": "",
                            "name": "sorter",
                            "type": "radio",
                            "value": "DATE",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "autocomplete": undefined,
                            "checked": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "autocomplete": undefined,
                            "checked": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                        },
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-toggle": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-toggle": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "sorter-buttons btn-toolbar",
          "role": "toolbar",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "sorter-button-group btn-group",
              "data-toggle": "buttons",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "sort-by-date btn btn-default active",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "autocomplete": "off",
                      "checked": "",
                      "name": "sorter",
                      "type": "radio",
                      "value": "DATE",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "release date",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autocomplete": undefined,
                      "checked": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autocomplete": undefined,
                      "checked": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "data": "release date",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "sorter",
                        "type": "radio",
                        "value": "DATE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "sort-by-rating btn btn-default",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "sorter",
                        "type": "radio",
                        "value": "RATING",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "ratings",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "ratings",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "sorter",
                          "type": "radio",
                          "value": "RATING",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "sort-by-rating btn btn-default",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "autocomplete": "off",
                      "name": "sorter",
                      "type": "radio",
                      "value": "RATING",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "ratings",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autocomplete": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autocomplete": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "data": "ratings",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "sorter",
                        "type": "radio",
                        "value": "RATING",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "sort-by-date btn btn-default active",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "sorter",
                        "type": "radio",
                        "value": "DATE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "release date",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "release date",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "sorter",
                          "type": "radio",
                          "value": "DATE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-toggle": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-toggle": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "Sort by",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
